molgenis-data-validation/src/main/java/org/molgenis/data/validation/EntityAttributesValidator.java

```diff
@@ -97,6 +114,58 @@
 		return violations;
 	}
 
+	private ConstraintViolation checkMref(Entity entity, AttributeMetaData attr, EntityMetaData meta)
+	{
+		Iterable<Entity> refEntities;
+		try
+		{
+			refEntities = entity.getEntities(attr.getName());
+		}
+		catch (Exception e)
+		{
+			return createConstraintViolation(entity, attr, meta, "Not a valid entity, expected an entity list.");
+		}
+		if (refEntities == null)
+		{
+			return createConstraintViolation(entity, attr, meta, "Not a valid entity, expected an entity list.");
+		}
+		for (Entity refEntity : refEntities)
+		{
+			if (refEntity == null)
+			{
+				return createConstraintViolation(entity, attr, meta, "Not a valid entity, null is not allowed");
+			}
```

Comment:
https://github.com/molgenis/molgenis/issues/4336#issuecomment-178492639 :
> The behavior for requesting a single entity by id is returning null. The behavior of requesting multiple entities by id is thus consistent and I propose to a) keep it that way b) revert the ElasticSearchRepository.findAll() fix c) mention this behavior in the DataService and Repository docs.

Didn't we decide that null *is* allowed/required when requesting multiple entities?